nucliadb_protos/nodereader.proto

```diff
@@ -186,6 +186,79 @@ message EntitiesSubgraphResponse {
 }
 
 
+message GraphQuery {
+    message Node {
+        optional string value = 1;
+        optional utils.RelationNode.NodeType node_type = 2;
+        optional string node_subtype = 3;
+        uint32 fuzzy_distance = 4;
+        bool as_prefix = 5;
+    }
+
+    message Relation {
+        optional string value = 1;
+    }
+
+    message Path {
+        optional Node source = 1;
+        optional Relation relation = 2;
+        optional Node destination = 3;
+        bool undirected = 4;
+    }
+
+    message BoolQuery {
+        repeated GraphQuery operands = 1;
+    }
+
+    oneof query {
+        // Search a node in any position
+        Node node = 1;
+        // Search for relations
+        Relation relation = 2;
+        // Search for directed/undirected paths. It can also express
+        // source/destination node queries
+        Path path = 3;
+
+    }
+}
+
+message GraphSearchRequest {
+    string shard = 1;
+    GraphQuery query = 2;
+
+    // decides what do we want in return: paths (triplets), nodes or relations.
+    // Returning paths implies returning nodes and relations.
+    enum QueryKind {
+        PATH = 0;
+        NODES = 1;
+        RELATIONS = 2;
+    }
+    QueryKind kind = 3;
```

Comment:
As far as I can see, this is unused and we always return everything.

It also means that setting `kind = NODES` and `top_k = 10` is kind of a lie, and it may return the same entity 10 times (since it always returns triplets).

Is this expected?

Reply:
Kind is not used yet. The repeated entities should be implemented with a custom collector